transmission_interface/test/transmission_interface_loader_test.cpp

```diff
@@ -390,6 +390,239 @@ TEST_F(TransmissionInterfaceLoaderTest, SuccessfulLoad)
+  ASSERT_NO_THROW(eff_jnt_iface->getHandle(info_diff.joints_.back().name_));
+  JointHandle pos_jnt_handle_diff1 = pos_jnt_iface->getHandle(info_diff.joints_.front().name_);
+  JointHandle vel_jnt_handle_diff1 = vel_jnt_iface->getHandle(info_diff.joints_.front().name_);
+  JointHandle eff_jnt_handle_diff1 = eff_jnt_iface->getHandle(info_diff.joints_.front().name_);
+  JointHandle pos_jnt_handle_diff2 = pos_jnt_iface->getHandle(info_diff.joints_.back().name_);
+  JointHandle vel_jnt_handle_diff2 = vel_jnt_iface->getHandle(info_diff.joints_.back().name_);
+  JointHandle eff_jnt_handle_diff2 = eff_jnt_iface->getHandle(info_diff.joints_.back().name_);
+
+  // Propagate state forward
+  act_pos.assign(3,  50.0);
+  act_vel.assign(3, -50.0);
+  act_eff.assign(3,   1.0);
+
+  act_to_jnt_state->propagate();
+
+  EXPECT_NEAR( 1.5, pos_jnt_handle_red.getPosition(),   EPS);
+  EXPECT_NEAR( 1.5, pos_jnt_handle_diff1.getPosition(), EPS);
+  EXPECT_NEAR( 0.5, pos_jnt_handle_diff2.getPosition(), EPS);
+
+  EXPECT_NEAR(-1.0, pos_jnt_handle_red.getVelocity(),   EPS);
+  EXPECT_NEAR(-1.0, pos_jnt_handle_diff1.getVelocity(), EPS);
+  EXPECT_NEAR( 0.0, pos_jnt_handle_diff2.getVelocity(), EPS);
+
+  EXPECT_NEAR( 50.0, pos_jnt_handle_red.getEffort(),     EPS);
+  EXPECT_NEAR(100.0, pos_jnt_handle_diff1.getEffort(),  EPS);
+  EXPECT_NEAR(  0.0, pos_jnt_handle_diff2.getEffort(),   EPS);
+
+  // Propagate position commands forward
+  pos_jnt_handle_red.setCommand(1.5);
+  pos_jnt_handle_diff1.setCommand(1.5);
+  pos_jnt_handle_diff2.setCommand(0.5);
+
+  jnt_to_act_pos_cmd->propagate();
+
+  EXPECT_NEAR(50.0, act_pos_cmd_handle_red.getPosition(),   EPS);
+  EXPECT_NEAR(50.0, act_pos_cmd_handle_diff1.getPosition(), EPS);
+  EXPECT_NEAR(50.0, act_pos_cmd_handle_diff2.getPosition(), EPS);
+
+  // Propagate velocity commands forward
+  vel_jnt_handle_red.setCommand(1.0);
+  vel_jnt_handle_diff1.setCommand(1.0);
+  vel_jnt_handle_diff2.setCommand(0.0);
+
+  jnt_to_act_vel_cmd->propagate();
+
+  EXPECT_NEAR(-50.0, act_vel_cmd_handle_red.getVelocity(),   EPS);
+  EXPECT_NEAR(-50.0, act_vel_cmd_handle_diff1.getVelocity(), EPS);
+  EXPECT_NEAR(-50.0, act_vel_cmd_handle_diff2.getVelocity(), EPS);
+
+  // Propagate effort commands forward
+  eff_jnt_handle_red.setCommand(50.0);
+  eff_jnt_handle_diff1.setCommand(100.0);
+  eff_jnt_handle_diff2.setCommand( 0.0);
+
+  jnt_to_act_eff_cmd->propagate();
+
+  EXPECT_NEAR(1.0, act_eff_cmd_handle_red.getEffort(), EPS);
+  EXPECT_NEAR(1.0, act_eff_cmd_handle_diff1.getEffort(), EPS);
+  EXPECT_NEAR(1.0, act_eff_cmd_handle_diff2.getEffort(), EPS);
+
+  // Now propegate things in the reverse direction
+  RawJointDataMap* joint_data_map = &trans_iface_loader.getData()->raw_joint_data_map;
+  joint_data_map->operator[]("foo_joint").position = 1.5;
+  joint_data_map->operator[]("bar_joint").position = 1.5;
+  joint_data_map->operator[]("baz_joint").position = 0.5;
+
+  joint_data_map->operator[]("foo_joint").velocity = -1.;
```

Comment:
I'd prefer adding the `0` after the `.` for these six floating point literals, for readability and consistency with the rest of this file.